gax-grpc/src/main/java/com/google/api/gax/grpc/OperationFuture.java

```diff
@@ -29,348 +29,58 @@
-    return initialOperationFuture.get().getMetadata();
-  }
-
-  /**
-   * Returns the Operation object returned from the call to start the Operation. Blocks if the
-   * initial call to start the Operation hasn't returned yet.
-   */
-  public final Operation getFirstOperationData() throws InterruptedException, ExecutionException {
-    return initialOperationFuture.get();
-  }
-
-  /** Awaits any ongoing asynchronous work. */
-  public final void awaitAsyncCompletion() throws InterruptedException {
-    asyncCompletionLatch.await();
-  }
-
-  /** Awaits any ongoing asynchronous work. */
-  public final void awaitAsyncCompletion(long timeout, TimeUnit unit) throws InterruptedException {
-    asyncCompletionLatch.await(timeout, unit);
-  }
-
-  private static <ResponseT extends Message> void setResultFromOperation(
-      SettableFuture<ResponseT> resultFuture, Operation operation, Class<ResponseT> responseClass) {
-    Status status = Status.fromCodeValue(operation.getError().getCode());
-    if (!status.equals(Status.OK)) {
-      String message =
-          "Operation with name \"" + operation.getName() + "\" failed with status = " + status;
-      resultFuture.setException(new ApiException(message, null, status.getCode(), false));
-    } else {
-      Any responseAny = operation.getResponse();
-      if (responseAny.is(responseClass)) {
-        ResponseT response;
-        try {
-          response = responseAny.unpack(responseClass);
-          resultFuture.set(response);
-        } catch (InvalidProtocolBufferException e) {
-          String message =
-              "Operation with name \""
-                  + operation.getName()
-                  + "\" succeeded, but encountered a problem unpacking it.";
-          resultFuture.setException(new ApiException(message, e, status.getCode(), false));
-        }
-      } else {
-        String message =
-            "Operation with name \""
-                + operation.getName()
-                + "\" succeeded, but it is not the right type; "
-                + "expected \""
-                + responseClass.getName()
-                + "\" but found \""
-                + responseAny.getTypeUrl()
-                + "\"";
-        resultFuture.setException(new ClassCastException(message));
-      }
-    }
-  }
-
-  private static boolean isCancelled(Operation operation) {
-    if (operation.getError() != null) {
-      Status status = Status.fromCodeValue(operation.getError().getCode());
-      return status.getCode().equals(Status.Code.CANCELLED);
-    }
-    return false;
+  public ResponseT get(long timeout, TimeUnit unit)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    return resultFuture.get(timeout, unit);
   }
 
-  static class Waiter {
-    public void wait(Duration duration) throws InterruptedException {
-      Thread.sleep(duration.toMillis());
-    }
+  public ApiFuture<Operation> getInitialFuture() {
```

Comment:
Please check the previous comment https://github.com/googleapis/gax-java/pull/315#discussion_r119444944